File: groupy_api/Api_groupy/WebApplication1/Repositories/CidadeRepository.cs
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;
using System.Linq;
using System.Threading.Tasks;
using WebApplication1.Interfaces;
using WebApplication1.Models.Localidade;
using WebApplication1.Tools.DataBase;

namespace WebApplication1.Repositories
{
    public class CidadeRepository : ICidadeRepository
    {
        private readonly Context _context;

        public CidadeRepository(Context context)
        {
            _context = context;
        }
        public async Task<List<Cidade>> GetAll()
        {
            return await _context.Cidade.ToListAsync();
        }

        public async Task<Cidade> GetById(int cidadeId)
        {
            return await _context.Cidade.FindAsync(cidadeId);
        }

        public async Task<List<Cidade>> GetByUf(string uf)
        {
            return await _context.Cidade.Where(g => g.Estado.Uf == uf).Include(c=> c.Estado).ToListAsync();
        }

        public async Task<Cidade> GetByUser(int userId)
        {
            var user = await _context.User.Where(x => x.Id == userId).Include(x => x.Cidade).FirstOrDefaultAsync();
            if(user.Cidade != null)
            {
                return user.Cidade;
            }
            return null;
        }
    }
}

File: groupy_appV3/groupy_app/lib/repositories/localidade/cidade/cidade_repository.dart
import 'package:groupy_app/application/rest_client/rest_client.dart';
import 'package:groupy_app/models/cidade_model.dart';
import 'package:groupy_app/repositories/localidade/cidade/icidade_repository.dart';

class CidadeRepository implements ICidadeRepository {
  @override
  Future<List<Cidade>> getAll() async {
    var response = await RestClient().dio.get('/Cidade');
    List<Cidade> listCidade = [];
    for (var json in response.data as List) {
      final cidade = Cidade.fromJson(json);
      listCidade.add(cidade);
    }
    return listCidade;
  }

  @override
  Future<List<Cidade>> getByUf(String uf) async {
    var response = await RestClient().dio.get('/Cidade/uf/' + uf);
    List<Cidade> listCidade = [];
    for (var json in response.data as List) {
      final cidade = Cidade.fromJson(json);
      listCidade.add(cidade);
    }
    return listCidade;
  }

  @override
  Future<Cidade?> getByUser(String userId) async {
    var response = await RestClient().dio.get('/Cidade/user' + userId);
    if (response.statusCode == 200) {
      return Cidade.fromJson(response.data);
    }
    return null;
  }

  @override
  Future<Cidade?> getOne(String id) async {
    var response = await RestClient().dio.get('/Cidade/' + id);
    if (response.statusCode == 200) {
      return Cidade.fromJson(response.data);
    }
    return null;
  }
}
